WikiClientLibrary/Generators/AllPagesGenerator.cs

```diff
@@ -110,4 +115,19 @@ public enum PropertyFilterOption
         /// </summary>
         WithoutProperty,
     }
+
+    public enum ProtectionType
+    {
+        Edit,
+        Move,
+        Upload
+    }
+
+    public enum ProtectionLevel
```

Comment:
Some wikis may have more level. I used the English Wikipedia as a reference, the enum can be extended later.

Reply:
I've checked DB table and [`pr_level`](https://www.mediawiki.org/wiki/Manual:Page_restrictions_table#pr_level) is a column of free text. Then probably you can use `string` as protection level, and use a another static class like `ProtectionLevels` to define common protection levels.

In the future we might introduce some utility structures like [`HttpMethod` class](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpmethod?view=net-8.0) for better strong-typing support.